modules/bitcoind.nix

```diff
@@ -328,21 +338,22 @@ in {
           ${extraRpcauth}
           ${/* Enable bitcoin-cli for group 'bitcoin' */ ""}
           printf "rpcuser=${cfg.rpc.users.privileged.name}\nrpcpassword="; cat "${secretsDir}/bitcoin-rpcpassword-privileged";
+          echo
+          ${optionalString (cfg.getPublicAddressCmd != "") ''
+            echo "externalip=$(${cfg.getPublicAddressCmd})"
+          ''}
         )
         confFile='${cfg.dataDir}/bitcoin.conf'
         if [[ ! -e $confFile || $cfg != $(cat $confFile) ]]; then
           install -o '${cfg.user}' -g '${cfg.group}' -m 640 <(echo "$cfg") $confFile
         fi
       '';
-      postStart = ''
-        # Poll until bitcoind accepts commands. This can take a long time.
-        while ! ${cfg.cli}/bin/bitcoin-cli getnetworkinfo &> /dev/null; do
-          sleep 1
-        done
-      '';
       serviceConfig = nix-bitcoin-services.defaultHardening // {
+        Type = "notify";
+        NotifyAccess = "all";
         User = "${cfg.user}";
         Group = "${cfg.group}";
+        TimeoutStartSec = 300;
```

Comment:
IIUC if the node is reindexing (i.e. started with `reindex=1`), we'll only get a notification after that is done. Reindexing is the same as initial sync without block download, so it usually takes hours.

Reply:
I guess we should set`TimeoutStartSec ="infinity"` then.

Reply:
I believe reindexing doesn't count as a part of startup sequence - the RPC is available during reindexing (you can check status at least). So `startupnotify` command should fire soon enough.

But will check it and reattempt my integration. Thanks for heads up!

Reply:
@Kixunil, you're right. I've checked the code; [AppInitMain()](https://github.com/bitcoin/bitcoin/blob/16b784d953365bb2d7ae65acd2b20a79ef8ba7b6/src/init.cpp#L1267) doesn't block on reindexing.
[Step 7: load block chain](https://github.com/bitcoin/bitcoin/blob/16b784d953365bb2d7ae65acd2b20a79ef8ba7b6/src/init.cpp#L1530), can take a few minutes, but `TimeoutStartSec = 300` should be sufficient. I've reverted the last change.

Reply:
I was thinking about checking the code but didn't have enough time, thanks a lot for doing it!